src/lib/core/ripple/ripple.ts

```diff
@@ -86,36 +77,71 @@ export class MatRipple implements OnChanges, OnDestroy {
    */
   @Input('matRippleSpeedFactor') speedFactor: number = 1;
 
-  /** Custom color for ripples. */
-  @Input('matRippleColor') color: string;
+  /**
+   * The element that triggers the ripple when click events are received.
+   * Defaults to the directive's host element.
+   */
+  @Input('matRippleTrigger')
+  get trigger() { return this._trigger; }
+  set trigger(trigger: HTMLElement | null) {
+    this._trigger = trigger;
+
+    // The events for a trigger element shouldn't be registered if ripples are disabled.
+    // As soon as the ripples are enabled again, the events for the current trigger will be added.
+    if (!this.disabled && this._isInitialized) {
+      this._rippleRenderer.setTriggerElement(trigger);
```

Comment:
It feels strange to call `setTriggerElement` when you really just want one of the _side-effects_ of `setTriggerElement`. Would it be easier to follow if there was a function to specifically add/remove the event listeners?

Reply:
What do you mean with side-effects? Maybe the function name is just not good and it should be called `registerTriggerEvents()` or something like that.

Reply:
Based on the name, the primary purpose of `setTriggerElement` would be the line
```ts
 this._triggerElement = element;
```

The event stuff is incidental to updating the trigger element. It be a bit better as something like
```ts
  /** Sets the trigger element and registers the mouse events. */
  setTriggerElement(element: HTMLElement | null) {
    if (this._triggerElement === element) {
      return;
    }

    this._removeTriggerListeners();
    this._triggerElement = element;
    this._addTriggerListeners();
  }
```

Then other call site that don't need to update to a new element just call `_removeTriggerListeners` and `_addTriggerListeners`

Reply:
I see what you mean, but the actual event registration should be the primary purpose of this function (or part in the `RippleRenderer`)

In that case I would just call the method `_setupTriggerEvents()` and keep the implementation as it is. The `_triggerElement` variable is just there for the event un-registration, and the real `trigger` element is stored in the `MatRipple` class. Does that work for you?

Reply:
Agreed that at least the naming should change to something involving "events".